src/test/java/org/javarosa/xpath/test/XPathEvalTest.java

```diff
@@ -99,6 +99,24 @@ public void testTypeCoercion(){
     }
 
     @Test
+    public void sortTests() {
+        // simple sort
+        testEval("sort('commcare is the best tool ever', false())", null, null, "tool the is ever commcare best");
+
+        // sort 2nd list by 1st
+        testEval("sort('2222 5555 9999 1111', 'd b c a', true())", null, null, "1111 5555 9999 2222");
+        testEval("sort('a b c d e f', '4 2 1 5 3 2', false())", null, null, "d a e f b c");
+        testEval("sort('c c z f z f', '4 2 1 5 3 2', true())", null, null, "z c f z c f");
```

Comment:
To test above, I'd include some examples where there are multiple items with matching sort orders int he second list, IE:

a b c d : 3 2 1 2 = c b d a OR c d b a